Portal/src/Datahub.Infrastructure/Services/DatahubTelemetryAuditingService.cs

```diff
@@ -79,12 +79,21 @@ public async Task TrackEvent(string name, params (string Key, string Value)[] de
 
     private async Task AppendIdentity(Dictionary<string, string> dictionary, bool anonymous = false)
     {
-        var user = anonymous ? await userInformationService.GetAnonymousGraphUserAsync() : await userInformationService.GetCurrentGraphUserAsync();
-        if (user != null)
+        try
         {
-            dictionary["userId"] = user.Id;
-            dictionary["userName"] = user.DisplayName;
-            dictionary["userPrincipalName"] = user.UserPrincipalName;
+            var user = anonymous ? await userInformationService.GetAnonymousGraphUserAsync() : await userInformationService.GetCurrentGraphUserAsync();
+            if (user != null)
+            {
+                dictionary["userId"] = user.Id;
+                dictionary["userName"] = user.DisplayName;
+                dictionary["userPrincipalName"] = user.UserPrincipalName;
+            }
+        }
+        catch (Exception ex)
+        {
+            dictionary["userId"] = "Unknown";
```

Comment:
if this is for the API - we might want to use one of the claims in the token as user id so that it is tracked